adds neonv7 for volk_32f_sin_32f

ast opened the pull request:
I have discussed the licensing of this code with @bhilburn.

Performance is about 2x on Raspberry Pi 4 in armv7 mode and 4x in aarch64 mode.

```
# Raspberry Pi 4
./volk_32f_sin_32f_test.sh 

RUN_VOLK_TESTS: volk_32f_sin_32f(131071,1)
generic completed in 8.54417 ms
neon completed in 3.71878 ms
Best aligned arch: neon
Best unaligned arch: neon
```

ast committed:
adds neonv7 for volk_32f_sin_32f

bhilburn commented:
The license of the code isn't a problem, here, and good job carrying the license notice, @ast. One thing: we need to make it clear what part of that file that license refers to. Just below the L24 copyright statements, can you add a note about what function(s) Julien Pommier holds copyright to?

You're welcome to keep your copyright line there, too, but fwiw it doesn't have any legal effect once upstreamed as your copyright is assigned to the FSF for these changes -- unless you specifically want to withhold that, here.

ast commented:
Ok will do that. Just put that copyright notice because I modified the original function heavily, maybe there's a better way saying that..?

bhilburn commented:
@ast - Ahh, I see. You did exactly the right thing, then =)

ast committed:
Add function name to copyright notice. Adds cos and tan for neon

ast commented:
Ok added the function name to the copyright notice @bhilburn.

Added _cos_ and _tan_ as well as they are closely related. Cleaned up the code slightly in the process. All this could probably be squashed into one commit...

It's interesting to note that these kernels were about 4x faster than generic on Raspberry Pi 3 aarch64 (without any modifications) and now they are 2x on Raspberry Pi 4 armv7.

I haven't had time to test aarch64 on Raspberry Pi 4 but I wonder if this difference is because generic is faster on Raspberry Pi 4 or if we will gain an additional 2x on aarch64. Aarch64 has twice as many NEON registers.


```
RUN_VOLK_TESTS: volk_32f_sin_32f(131071,1)
generic completed in 3.44148 ms
neon completed in 1.47252 ms
Best aligned arch: neon
Best unaligned arch: neon

RUN_VOLK_TESTS: volk_32f_cos_32f(131071,1)
neon completed in 1.54166 ms
generic_fast completed in 10.0799 ms
generic completed in 3.60755 ms
Best aligned arch: neon
Best unaligned arch: neon

RUN_VOLK_TESTS: volk_32f_tan_32f(131071,1)
neon completed in 2.06868 ms
generic completed in 9.07185 ms
Best aligned arch: neon
Best unaligned arch: neon
```

michaelld reviewed (approved):
I have no objection to adding in this Neon-specific kernel, even though I can't test it at the moment. We have to have some trust in the developers that they are doing checks on their own hardware, yes?

jdemel reviewed (approved):
LGTM

michaelld commented:
Some recent merge has caused a conflict. @ast Please rebase & fix the conflict & then I'll merge.

michaelld commented:
Ping ... can we get a rebase here so that this can be merged?

ast commented:
Sorry, I've been away. I will rebase.

michaelld commented:
Ping again ... can we get a rebase here so that this can be merged? It's ready to go beyond rebasing, and we'd like to get this into the next release coming soon-ish ...

ast commented:
Hi @michaelld, sorry been quite busy! I'll try to do it today!

Very happy about your hard work merging in ARM PRs. Must have been quite a boost for some flowgraphs.

ast commented:
I created shiny new and improved PR. Please merge that one and delete this one.

https://github.com/gnuradio/volk/pull/314

michaelld commented:
acknowledge
